In [ ]:
from google.colab import drive
drive.mount('/content/drive')


Go to this URL in a browser: https://accounts.google.com/o/oauth2/auth?client_id=947318989803-6bn6qk8qdgf4n4g3pfee6491hc0brc4i.apps.googleusercontent.com&redirect_uri=urn%3Aietf%3Awg%3Aoauth%3A2.0%3Aoob&scope=email%20https%3A%2F%2Fwww.googleapis.com%2Fauth%2Fdocs.test%20https%3A%2F%2Fwww.googleapis.com%2Fauth%2Fdrive%20https%3A%2F%2Fwww.googleapis.com%2Fauth%2Fdrive.photos.readonly%20https%3A%2F%2Fwww.googleapis.com%2Fauth%2Fpeopleapi.readonly&response_type=code

Enter your authorization code:
··········
Mounted at /content/drive


In [ ]:
import pandas as pd
import numpy as np
import matplotlib.pyplot as plt
import matplotlib as mpl    
from pandas import DataFrame

fig = plt.figure()
from numpy.random import randn


<Figure size 432x288 with 0 Axes>

In [ ]:
upper5 = pd.read_csv('/content/drive/My Drive/소통형도서관_프로젝트/자료/UPPER5.csv') 
so = pd.read_csv('/content/drive/My Drive/대출목록.csv',sep="," )
best = pd.read_csv('/content/drive/My Drive/소통형도서관_프로젝트/자료/베스트_이용도서.csv',sep=",")


In [ ]:
# 일단 import 
%matplotlib inline  

import matplotlib as mpl  # 기본 설정 만지는 용도
import matplotlib.pyplot as plt  # 그래프 그리는 용도
import matplotlib.font_manager as fm  # 폰트 관련 용도

# 현재 설정되어 있는 폰트 사이즈와 글꼴을 알아보자
!python --version
def current_font():
  print(f"설정 폰트 글꼴: {plt.rcParams['font.family']}, 설정 폰트 사이즈: {plt.rcParams['font.size']}")  # 파이썬 3.6 이상 사용가능하다
        
current_font()

Python 3.6.8
설정 폰트 글꼴: ['sans-serif'], 설정 폰트 사이즈: 10.0


In [ ]:
!apt-get update -qq
!apt-get install fonts-nanum* -qq

Selecting previously unselected package fonts-nanum.
(Reading database ... 131183 files and directories currently installed.)
Preparing to unpack .../fonts-nanum_20170925-1_all.deb ...
Unpacking fonts-nanum (20170925-1) ...
Selecting previously unselected package fonts-nanum-eco.
Preparing to unpack .../fonts-nanum-eco_1.000-6_all.deb ...
Unpacking fonts-nanum-eco (1.000-6) ...
Selecting previously unselected package fonts-nanum-extra.
Preparing to unpack .../fonts-nanum-extra_20170925-1_all.deb ...
Unpacking fonts-nanum-extra (20170925-1) ...
Selecting previously unselected package fonts-nanum-coding.
Preparing to unpack .../fonts-nanum-coding_2.5-1_all.deb ...
Unpacking fonts-nanum-coding (2.5-1) ...
Setting up fonts-nanum-extra (20170925-1) ...
Setting up fonts-nanum (20170925-1) ...
Setting up fonts-nanum-coding (2.5-1) ...
Setting up fonts-nanum-eco (1.000-6) ...
Processing triggers for fontconfig (2.12.6-0ubuntu2) ...


In [ ]:
path = '/usr/share/fonts/truetype/nanum/NanumGothic.ttf'  # 설치된 나눔글꼴중 원하는 녀석의 전체 경로를 가져오자
font_name = fm.FontProperties(fname=path, size=15).get_name()
print(font_name)
plt.rc('font', family=font_name)

NanumGothic


In [ ]:
mpl.rcParams['axes.unicode_minus'] = False

In [ ]:
print(best.loc[0])

등록번호                                         778967
서명      나는 나로 살기로 했다 : 냉담한 현실에서 어른살이를 위한 to do list
청구기호                                     818.7 김57나
저자                                              김수현
출판사                                            마음의숲
출판년도                                           2016
유형                                            818.7
이용빈도                                             20
Name: 0, dtype: object


In [ ]:
bestb = best.loc[[0,1,2,3,4,5,6,7,8,9,10,11,12,13,14,15,16,17,18,19,20,21,22,23,24,25,26,27,28,29,30,31,32,33,34,35,36,37,38,39,40,41,42,43,44,45,46,47,48,49,50,51,52,53,54,55,54,55,56,57,58,59,60]]
bestb.tail(5)

등록번호                              서명  ...       유형 이용빈도
56  778007  청춘을 위한 나라는 없다 청년논객 한윤형의 잉여탐구생활  ...  305.242   11
57  797549                  트렌드 코리아 . 2019  ...  330.951   11
58  797955           12가지 인생의 법칙 : 혼돈의 해독제  ...    179.9   11
59  793774            한때 소중했던 것들 : 이기주 산문집  ...    818.7   11
60  800470             아가씨와 밤 : 기욤 뮈소 장편소설  ...  843.914   11

[5 rows x 8 columns]

In [ ]:
upper5_loc = upper5.loc[[0, 2, 5, 6]]
upper5_loc

등록번호  ...                       566731
0    서명  ...  1리터의 눈물: 불치병과 싸우는 소녀 아야의 일기
2    저자  ...                        키토 아야
5    유형  ...                       895.66
6  이용빈도  ...                            5

[4 rows x 1706 columns]

In [ ]:
best50 = best.iloc[0:50, :]
best50_3 = best50.loc[:,['서명','이용빈도','유형']]
best50_3.head(6)

서명  이용빈도       유형
0  나는 나로 살기로 했다 : 냉담한 현실에서 어른살이를 위한 to do list    20    818.7
1                   돌이킬 수 없는 약속 : 야쿠마루 가쿠 장편소설    20  895.636
2                          82년생 김지영 : 조남주 장편소설    19    813.7
3                  죽고 싶지만 떡볶이는 먹고 싶어 : 백세희 에세이    17    818.7
4                               아몬드 : 손원평 장편소설    16    813.7
5                                       언어의 온도    16    818.7

In [ ]:
print(best50_3[best50_3['서명'].isin(['언어의 온도'])])

        서명  이용빈도     유형
5   언어의 온도    16  818.7
10  언어의 온도    15  818.7
28  언어의 온도    13  818.7


#지난 2018-06-01 ~ 2019-09-30 까지의 대출데이터

In [ ]:
so.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 29951 entries, 0 to 29950
Data columns (total 6 columns):
학과      29951 non-null object
등록번호    29951 non-null object
청구기호    29951 non-null object
서명      29951 non-null object
저자      29907 non-null object
대출일     29951 non-null object
dtypes: object(6)
memory usage: 1.4+ MB


In [ ]:
so_1 = so.iloc[:, :]
so_2 = so.loc[:,['학과','청구기호','서명','대출일']]
so_2.info(6)

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 29951 entries, 0 to 29950
Data columns (total 4 columns):
학과      29951 non-null object
청구기호    29951 non-null object
서명      29951 non-null object
대출일     29951 non-null object
dtypes: object(4)
memory usage: 936.0+ KB


##공과대학 대여권수

In [ ]:
so_eg = so_2[so_2['학과'].isin(['산업경영공학과','컴퓨터공학과','정보통신공학과','멀티미디어공학전공','멀티미디어학부','멀티미디어영상전공',
                                   '토목·건축공학부','미디어영상전공','전기전자공학과' ,'건축공학전공',
                                   '토목환경공학전공','기계공학과','화학공학과','컴퓨터무인통신기술학과'])]

#신규인'신소재공학과','스마트융합공학부'제외 , 재학생이 없을것으로 판단 '공학교육 혁신센터 제외'
so_eg.shape

(1344, 4)

##사범대학 대여권수

In [ ]:
so_edu = so_2[so_2['학과'].isin(['국어교육과','영어교육과','교육학과','역사교육과','미술교육과','수학교육과'])]
#교육연수원, 교직부 제외
so_edu.shape

(2976, 4)

##문과대학 대여권수

In [ ]:
so_la =  so_2[so_2['학과'].isin(['국어국문·창작학과','영어영문학과','외국어문학부','일어일문학전공','프랑스어문학전공',
                                    '문헌정보학과','사학과','기독교학과','인문학부 철학상담학전공','일본·프랑스어문학과'])]
so_la.shape

(4758, 4)

##경상대학 대여권수

In [ ]:
so_ec =  so_2[so_2['학과'].isin(['경영학과','회계학과','무역학과','경제학부','경제학전공',
                                    '중국경제통상학전공','컨벤션호텔경영학과','비즈니스통계학과'])]
#방촌국제화 훈련단 제외
so_ec.shape

(2395, 4)

##법정대학 대여권수

In [ ]:
so_ss =  so_2[so_2['학과'].isin(['법학부','행정·경찰학부' '행정학전공','경찰학전공 모집전공','정치·언론학과' ,'사회복지학과','아동복지학과','도시부동산학과'])]
so_ss.shape

(1778, 4)

##생명나노과학대학 대여권수

In [ ]:
so_ls =  so_2[so_2['학과'].isin(['생명시스템과학과' ,'식품영양학과','화학과' ,'수학과','간호학과','스포츠과학과','화공신소재공학과'])]
so_ls.shape

(2525, 4)

##사회적경제융합대학 대여권수

In [ ]:
so_se =  so_2[so_2['학과'].isin(['사회복지학과','아동복지학과'])]
                               
#상담심리학과 , 사회적경제기업학과 , 빅데이터응용학과 신규 생성학과 제외
so_se.shape

(820, 4)

##린튼글로벌비지니스 대여권수

In [ ]:
so_gb =  so_2[so_2['학과'].isin(['글로벌비즈니스전공','글로벌IT경영전공'])]
so_gb.info()

<class 'pandas.core.frame.DataFrame'>
Int64Index: 0 entries
Data columns (total 4 columns):
학과      0 non-null object
청구기호    0 non-null object
서명      0 non-null object
대출일     0 non-null object
dtypes: object(4)
memory usage: 0.0+ bytes


##조형예술학부 대여권수

In [ ]:
so_art =  so_2[so_2['학과'].isin(['조형예술학부','융합디자인전공','회화전공','의류학전공'])]
so_art.shape

(860, 4)

##자유전공학부 대여권수

In [ ]:
so_free =  so_2[so_2['학과'].isin(['자유전공학부'])]
so_free.shape

(60, 4)

1)so_eg #공과대

2)so_edu #사범대

3)so_la #문과대

4)so_ec #경상대학

5)so_ss #법정대학

6)so_ls #생명나노대학

7)so_se #사회적경제융합대학

8)so_gb #린튼글로벌비지니스 - 값이 없음 

9)so_art #조형예술학부

10)so_free #자유전공학부


In [ ]:
so_2.tail(6) 

학과  ...       대출일
29945  LINC+사업단  ...  20190823
29946  LINC+사업단  ...  20190823
29947  LINC+사업단  ...  20190823
29948  LINC+사업단  ...  20190715
29949  LINC+사업단  ...  20190715
29950  LINC+사업단  ...  20190715

[6 rows x 4 columns]

위 결과에 등장하듯 링크사업단이나 신규모집 학과, 학생이 소속하지 않은 대학부, 다전공으로 이루어진 DH-SCHOOL 등등

실질적 홍보대상이 아닌 대상은 분석에서 제외 

In [ ]:
from pandas import DataFrame
so_maj = pd.concat([so_eg,so_edu,so_la,so_ec ,so_ss,so_ls , so_se,so_gb , so_art, so_free] )
so_maj1=so_maj.reset_index()


전체 29951권에서 실질적으로 재학생이 빌려간 권수는 17516권 뿐임을 알 수 있다.

In [ ]:
so_edu.head(5)

학과             청구기호               서명       대출일
785  교육학과       813.6 계66ㅂ      별을 헨다 ; 創作集  20180907
786  교육학과  370.193 R363sKㄱ          人間없는 학교  20180906
787  교육학과      813.6 최695과               廣場  20181113
788  교육학과     720.951 류14ㅇ  우리 옛 건축에 담긴 표정들  20181031
789  교육학과  891.733 N117ℓKㄱ              롤리타  20181129

In [ ]:
del so_maj1['index']

In [ ]:
so_maj1.head(3)

학과  ...       대출일
0  기계공학과  ...  20180816
1  기계공학과  ...  20181224
2  기계공학과  ...  20180921

[3 rows x 4 columns]

In [ ]:
so_maj2 = so_maj1.drop_duplicates(['서명'],keep = False)
so_maj2.head(5)

학과  ...       대출일
0  기계공학과  ...  20180816
1  기계공학과  ...  20181224
2  기계공학과  ...  20180921
4  기계공학과  ...  20180917
5  기계공학과  ...  20181024

[5 rows x 4 columns]

In [ ]:
import numpy as np
import pandas as pd
#데이터 프레임 리스트로 변환
#무쓸모 안됌 ㅋ

def mkRefdS(pre, roundN):
    re=[pre.loc[i, j] for i in pre.index for j in pre.columns]
    return(np.round(re, roundN))

In [ ]:
print(type(so_maj1['서명']))

<class 'pandas.core.series.Series'>


In [ ]:
so_maj_book =so_maj1['서명'].tolist() # 특정 칼럼만 리스트로 출력 

In [ ]:
w_count= {}
lists= so_maj_book
for lst in lists:
    try: w_count[lst]+= 1
    except: w_count[lst]=1
w_count
#리스트안의 중복된 값 개수확인

{'Robot analysis : the mechanics of serial and parallel manipulators': 1,
 '홋카이도=    Hokkaido': 1,
 '퀴디치의 역사': 1,
 '상실의 시대': 11,
 '(초심자를 위한)오실로스코프': 1,
 '부자아빠의 진실게임': 1,
 'ANSYS Workbench 왕초보 탈출하기': 3,
 '내 자아를 버려라': 1,
 '대한민국 20대, 펀드투자에 미쳐라': 1,
 '군주론': 47,
 '일 잘하는 사람 일 못하는 사람': 1,
 '내연기관': 1,
 '난장이가 쏘아올린 작은 공 : 조세희 소설집': 6,
 '(매트 리들리의) 붉은 여왕': 1,
 '매트랩 개요와 응용': 1,
 '(리빙토픽) 영양과 식생활': 2,
 '예수님처럼': 3,
 '돈 버는 사람은 분명 따로 있다': 1,
 '물은 답을 알고 있다': 1,
 '流體力學演習': 1,
 '신비한 동물사전': 4,
 '자동차공학개론 : 기초에서부터 새로운 시스템까지': 1,
 'CNC 프로그래밍 ＆ 가공기술': 1,
 '정상의 법칙 : 정상에 앉아 있는 사람은 뭔가 특별한 삶의 원리가 있다': 1,
 '(아하!) 물리수학': 1,
 '주거의 문화적 의미': 9,
 '파인만 씨, 농담도 잘하시네. 2': 1,
 '유체기계': 1,
 '유한요소 해석의 기초': 1,
 '먹어라 그렇지 않으면 먹힌다': 1,
 '(최신) KS 규격데이터북': 2,
 'Start with no = 상대방을 사로잡는 협상의 기술 /   노로 시작하라': 1,
 '내면세계의 질서와 영적성장': 1,
 '(Visual C++ 2008) MFC 윈도우 프로그래밍': 1,
 '고급영양학=    Advanced nutrition': 1,
 '사람풍경: 김형경 심리 여행 에세이': 1,
 '브루투스의 심장: 완전범죄 살인 릴레이': 3,
 '나는 희망의 증거가 되고 싶다 : 가발공장에서 하버드까지': 1,
 '비즈니스 협상론': 1,
 '자동차공학개론': 2,
 '(허브